aio_pika/abc.py

```diff
@@ -0,0 +1,672 @@
+import asyncio
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+from enum import Enum, IntEnum, unique
+from types import TracebackType
+from typing import (
+    Any, AsyncContextManager, Awaitable, Callable, Dict, FrozenSet, Generator,
+    Iterable, MutableMapping, NamedTuple, Optional, Set, Tuple, Type, TypeVar,
+    Union,
+)
+
+import aiormq
+from aiormq.abc import ExceptionType
+from pamqp.common import Arguments
+from yarl import URL
+
+from .pool import PoolInstance
+from .tools import CallbackCollection
+
+
+TimeoutType = Optional[Union[int, float]]
+
+NoneType = type(None)
+DateType = Union[int, datetime, float, timedelta, None]
+ExchangeParamType = Union["AbstractExchange", str]
+ConsumerTag = str
```

Comment:
Yes, it's actually the `TypeAlias` but `typing.TypeAlias` is available only since python 3.10.

Reply:
For example, in AMQP 0.9 specification might be clarified how the consumer-tag string is formed, then it will no longer be just a string. This is an alias for the future so that the user uses an explicitly specified type instead of a string in this particular case.